src/web/app/pages-instructor/instructor-student-activity-logs/instructor-student-activity-logs.component.ts

```diff
@@ -268,6 +269,130 @@ export class InstructorStudentActivityLogsComponent implements OnInit {
     const publishedDate: Date = new Date(publishedTime);
     const notViewedSince = publishedDate.getTime();
 
+    const filteredStudents = this.students
+        .filter((student: Student) => {
+          if (student.email === '') {
+            return false;
+          }
+
+          if (this.formModel.studentEmail !== '' && student.email !== this.formModel.studentEmail) {
+            return false;
+          }
+
+          if (this.formModel.showInactions && this.formModel.showActions) {
+            return true;
+          }
+
+          const studentKey = this.getStudentKey(log, student.email);
+
+          if (studentKey in this.studentToLog) {
+            if (this.formModel.showInactions) {
+              return false;
+            }
+          } else if (this.formModel.showActions) {
+            return false;
+          }
+
+          return true;
+        });
+
+    const logRowsData: SortableTableCellData[][] = filteredStudents.map((student: Student) => {
+      let status: string;
+      let entry;
+      let dataStyle: string = 'font-family:monospace; white-space:pre;';
+      let statusPrefix = this.logTypeToActivityDisplay(this.formModel.logType);
```

Comment:
I think we can use `let statusPrefix = this.logTypeToActivityDisplay(entry.feedbackSessionLogType);` if `studentKey in this.studentToLog`, otherwise we use `this.logTypeToActivityDisplay(this.formModel.logType);` for inactivities. This could get rid of the logic from line 330-395

Reply:
hey hieu, thank you for reviewing this PR! lines 330-395 adds the latest log for either access or submission activity type.

this addition is based on what's added in the if clause `studentKey in this.studentToLog`.

if we are initially adding an access activity type, in lines 330-395, we will search through the logs for this student and get the latest submission activity log, and vice versa.

Reply:
Can we move this logic to the backend instead? So we don't query for irrelevant logs on the frontend

Reply:
i understand that this chunk of codes is lengthy on the frontend. i checked and saw that we query all the logs based on the params we added when calling the logs service. in turn, it returns all the relevant logs. may i ask how should we move this logic to the backend? currently, we pull all logs based on the params we specify.

**here below is the current change in the PR:**

**for context:**
for our case, the edge case/enhancement, we'd want to display both `access` and `submission` logs. however, the initial implementation shows only the latest of either.

i did not want to affect the current implementation as i thought it was a really smart way to accommodate for inactivities too whenever specified in the control panel.

so, i used the logs that were returned from the log service and naively pushed the missing log into `logRowsData`, as this seems to be the only 'edge' case as the rest are well handled.

Reply:
@domlimm Let's put this on hold until this is merged first, then it will be much easier for you
https://github.com/TEAMMATES/teammates/pull/11857

Reply:
noted! thanks hieu! let me look at the PR again in the meantime!